jetstream/config.py

```diff
@@ -62,64 +68,91 @@
 @attr.s(auto_attribs=True)
 class Platform:
     config_spec_path: PathLike
-    metrics_module: Optional[ModuleType]
-    segments_module: Optional[ModuleType]
     enrollments_query_type: str
     # app_id to use to validate Outcomes.
     validation_app_id: str
+    metrics_module: Optional[ModuleType] = attr.ib(default=None)
+    segments_module: Optional[ModuleType] = attr.ib(default=None)
 
 
-PLATFORM_CONFIGS = {
-    "firefox_desktop": Platform(
-        Path(__file__).parent / "config" / "default_metrics.toml",
-        mozanalysis.metrics.desktop,
-        mozanalysis.segments.desktop,
-        "normandy",
-        "firefox-desktop",
-    ),
-    "fenix": Platform(
-        Path(__file__).parent / "config" / "fenix.toml",
-        mozanalysis.metrics.fenix,
-        None,
-        "glean-event",
-        "org.mozilla.fenix",
-    ),
-    "firefox_ios": Platform(
-        Path(__file__).parent / "config" / "firefox_ios.toml",
-        mozanalysis.metrics.firefox_ios,
-        None,
-        "glean-event",
-        "org.mozilla.ios.FirefoxBeta",
-    ),
-    "focus_android": Platform(
-        Path(__file__).parent / "config" / "focus_android.toml",
-        mozanalysis.metrics.focus_android,
-        None,
-        "glean-event",
-        "org.mozilla.focus",
-    ),
-    "klar_android": Platform(
-        Path(__file__).parent / "config" / "klar_android.toml",
-        mozanalysis.metrics.klar_android,
-        None,
-        "glean-event",
-        "org.mozilla.klar",
-    ),
-    "focus_ios": Platform(
-        Path(__file__).parent / "config" / "focus_ios.toml",
-        mozanalysis.metrics.focus_ios,
-        None,
+PARENT_DIR = Path(__file__).parent
+CONFIG_DIRECTORY = PARENT_DIR / "config"
+
+
+
+def _generate_platform_config(config: MutableMapping[str, Any]) -> Dict[str, Platform]:
+    """
+    Takes platform configuration and generates platform object map
+    """
+
+    valid_modules_metrics = mozanalysis.metrics.__all__
+    valid_modules_segments = mozanalysis.segments.__all__
```

Comment:
I published a new `mozanalysis` version which has all the packages in `mozanalysis.__all__`

Reply:
Amazing, thanks a lot for your help @scholtzan 🙇